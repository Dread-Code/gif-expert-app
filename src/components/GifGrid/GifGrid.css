.gif-grid {
  padding: 0;
  margin: 0;
  height: 100vh;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(min(100%, 35rem), 1fr));
  grid-template-rows: repeat(auto-fit, minmax(min(100%, 25em), 1fr));
  grid-gap: 0.4em;
  grid-auto-rows: minmax(25em, 1fr);
  grid-auto-flow: row dense;
  grid-auto-columns: minmax(35em, 1fr);
}
