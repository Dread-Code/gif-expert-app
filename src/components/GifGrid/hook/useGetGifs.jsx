import { useState, useEffect } from 'react'
import { getGifs } from '../../../helpers/getGifts'

const useGetGifs = category => {
  const [images, setImages] = useState([])
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    if (category)
      getGifs(category).then(imgs => {
        setImages(imgs)
        setLoading(true)
      })
  }, [category])
  return {
    images,
    loading
  }
}

export default useGetGifs
