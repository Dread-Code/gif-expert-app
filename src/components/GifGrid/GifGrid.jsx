import React from 'react'
import GifGridItem from '../GifGridItem'
import useGetGifs from './hook/useGetGifs'
import './GifGrid.css'

function GifGrid({ category }) {
  const { images, loading } = useGetGifs(category)

  if (!loading) return <h1 className="animate__animated animate__flash">Cargando...</h1>
  return (
    <>
      {images.map(img => (
        <GifGridItem {...img} key={img.id} />
      ))}
    </>
  )
}

export default GifGrid
