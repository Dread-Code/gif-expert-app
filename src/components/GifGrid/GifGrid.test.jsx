import { shallow } from 'enzyme'
import GifGrid from './GifGrid'
import useGetGifs from './hook/useGetGifs'

// pasamos jest  mock el custom hook
jest.mock('./hook/useGetGifs')

describe('GifGrid test', () => {
  let category = 'Kaneki'
  test('should run snapshot', () => {
    // para con la funcion que estamos llamando retornar el mock
    useGetGifs.mockReturnValue({
      images: [],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show items when the images are loaded', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa'
      }
    ]

    useGetGifs.mockReturnValue({
      images: gifs,
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })

  test('should show items when the loading is false', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa'
      }
    ]

    useGetGifs.mockReturnValue({
      images: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGridItem').length).toBe(0)
  })
})
