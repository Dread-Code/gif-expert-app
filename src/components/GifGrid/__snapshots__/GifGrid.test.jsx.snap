// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test should run snapshot 1`] = `<Fragment />`;

exports[`GifGrid test should show items when the images are loaded 1`] = `
<Fragment>
  <GifGridItem
    id="ABC"
    key="ABC"
    title="Cualquier cosa"
    url="https://localhost/cualquier/cosa.jpg"
  />
</Fragment>
`;

exports[`GifGrid test should show items when the loading is false 1`] = `
<h1
  className="animate__animated animate__flash"
>
  Cargando...
</h1>
`;
